File: Python/Cola.py
class Cola:
    
    def __init__(self):
        self.__items = []
    
    def encolar(self, dato):
        self.__items.append(dato)
    
    def desencolar(self):
        try:
            return self.__items.pop(0)
        except:
            return ValueError('La cola se encuentra vacia')
    
    def esVacia(self):
        if self.__items.count == 0:
            return True
        else:
            return False

File: C#/Arbol Binario/Arbol Binario/Cola.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Arbol_Binario
{
    public class Cola<T>
    {
        private List<T> datos = new List<T>();

        public void encolar(T elemento)
        {
            this.datos.Add(elemento);
        }

        public T desencolar()
        {
            T elementoTemporal = this.datos[0];
            this.datos.RemoveAt(0);
            return elementoTemporal;
        }

        public T tope()
        {
            return this.datos[0];
        }

        public bool esVacia()
        {
            if (this.datos.Count == 0)
            {
                return true;
            }
            else
            {
                return false;
            }
        }
    }
}
